internal/connor/all.go

```diff
@@ -0,0 +1,105 @@
+package connor
+
+import (
+	"github.com/sourcenetwork/defradb/client"
+
+	"github.com/sourcenetwork/immutable"
+)
+
+// all is an operator which allows the evaluation of
+// a number of conditions over a list of values
+// matching if all of them match.
+func all(condition, data any) (bool, error) {
+	switch t := data.(type) {
+	case []string:
```

Comment:
todo: I think it's better to use reflection here to avoid repetitive code:
```go
func all(condition, data any) {
    val := reflect.ValueOf(v)
    if val.Kind() != reflect.Slice {
        return false, client.NewErrUnhandledType("data", data)
    }

    for i := 0; i < val.Len(); i++ {
        m, err := eq(condition, val.Index(i).Interface())
            if err != nil {
                return false, err
            } else if !m {
                return false, nil
            }
        }
    }
    return true, nil
}
```
Or at least use generic:
```go
func allInArr[T any](slice []T, condition any) (bool, error) {
	for _, item := range slice {
		m, err := eq(condition, item)
		if err != nil {
			return false, err
		}
		if !m {
			return false, nil
		}
	}
	return true, nil
}

func all(condition, data any) (bool, error) {
	switch t := data.(type) {
	case []string:
		return allInArr(t, condition)
	case []immutable.Option[string]:
		return allInArr(t, condition)
	case []int64:
		return allInArr(t, condition)
// ...
```
This will also improve test coverage.

Reply:
reflect is a bit slow and not fully supported in wasm so lets try to avoid it for now.

Reply:
100% agree with Fred, we should avoid reflect, especially in a performance sensitive area like this.  Extracting the shared logic to a generic func would be good though.

Reply:
done